Sources/PointFree/Database.swift

```diff
@@ -201,20 +226,74 @@ private func fetchSubscriptionTeammates(ownerId: Database.User.Id) -> EitherIO<E
   )
 }
 
-private func updateUser(withId id: Database.User.Id, name: String, email: EmailAddress) -> EitherIO<Error, Prelude.Unit> {
+private func updateUser(
+  withId userId: Database.User.Id,
+  name: String?,
+  email: EmailAddress?,
+  emailSettings: [Database.EmailSetting.Newsletter]?
+  ) -> EitherIO<Error, Prelude.Unit> {
+
   return execute(
     """
     UPDATE "users"
-    SET "name" = $1, "email" = $2
+    SET "name" = COALESCE($1, "name"),
+        "email" = COALESCE($2, "email")
     WHERE "id" = $3
     """,
     [
       name,
-      email.unwrap,
-      id.unwrap.uuidString,
+      email?.unwrap,
+      userId.unwrap.uuidString,
     ]
     )
+    .flatMap(const(updateEmailSettings(settings: emailSettings, forUserId: userId)))
+}
+
+private func registerUser(withGitHubEnvelope envelope: GitHub.UserEnvelope) -> EitherIO<Error, Database.User?> {
+  return upsertUser(withGitHubEnvelope: envelope)
```

Comment:
`registerUser` does `upsertUser` followed by subscribing them to all our newsletters...